pages/src/route/router-service.ts

```diff
@@ -0,0 +1,48 @@
+import {memoize} from '@exadel/esl/src/modules/esl-utils/decorators';
+import {ESLEventUtils} from '@exadel/esl/src/modules/esl-event-listener/core';
+
+export class ESLDemoRouterService {
+  public contentSelector = '#content-routable';
+
+  public previousUrl: URL;
+
+  @memoize()
+  public static init(): ESLDemoRouterService {
+    const router = new ESLDemoRouterService();
+    router.previousUrl = new URL(location.href);
```

Comment:
Location is similar to URL object. Maybe it is enough to use a Location object?
or simplify
```suggestion
    router.previousUrl = new URL(location);
```